Lesson15/baljan/script.js

```diff
@@ -0,0 +1,361 @@
+  if (score >= 90) {
+    return "A";
+  } else if (score >= 80 && score < 90) {
+    return "B";
+  } else if (score >= 70 && score < 80) {
+    return "C";
+  } else if (score >= 60 && score < 70) {
+    return "D";
+  } else {
+    return "F";
+  }
+}
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+function replaceCharacter(str, oldChar, newChar) {
+  let newWord = word.replaceAll(oldChar, newChar);
+  return newWord;
+}
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence) {
+  return sentence
+    .split(" ")
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+    .join(" ");
+}
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop"
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+function trafficLight(color) {
+  switch (color) {
+    case "red":
+      console.log("Stop");
+      break;
+    case "yellow":
+      console.log("Caution");
+      break;
+    case "green":
+      console.log("Go");
+      break;
+    default:
+      console.log("Invalid color");
+  }
+}
+/*
+21. Check String Length (if-else)
+   - Define a function `isLongString(str)` that checks if the string length
+     is more than 10.
+   - Log "Long string" or "Short string".
+*/
+function isLongString(str) {
+  if (str.length > 10) {
+    console.log("Long string");
+  } else {
+    console.log("Short string");
+  }
+}
+/*
+22. Convert to Lowercase Then Check
+   - Define a function `isSpam(text)` that converts the text to lowercase
+     and checks if it includes "spam".
+   - If it does, log "This text is spam."
+   - Otherwise, log "This text is not spam."
+*/
+function isSpam(text) {
+  let lowerText = text.toLowerCase();
+  if (lowerText.includes("spam")) {
+    console.log("This text is spam!");
+  } else {
+    console.log("This text is not spam.");
+  }
+}
+/*
+23. Extract Initials
+   - Define a function `getInitials(fullName)` that uses .split() to get each name part,
+     then logs the capitalized first letter of each.
+   - Example: "John Doe" -> "J.D."
+*/
+function getInitials(fullName) {
+  return fullName
+    .split(" ")
+    .map(word => word[0].toUpperCase())
+    .join(".") + ".";
+}
+/*
+24. Switch: Month to Season
+   - Define a function `getSeason(monthNum)` (1-12). Use switch or if-else:
+     - 12, 1, 2  -> "Winter"
+     - 3, 4, 5   -> "Spring"
+     - 6, 7, 8   -> "Summer"
+     - 9, 10, 11 -> "Autumn"
+   - Log the season or "Invalid month" if out of range.
+*/
+function getSeason(monthNum) {
```

Comment:
In cases like this, using a switch statement is preferred. It will make your code easier to read. 
No need to change now. 🙂